codegen/src/main/resources/software/amazon/awssdk/codegen/rules/AwsEndpointProviderUtils.java.resource

```diff
@@ -150,19 +150,38 @@ public final class AwsEndpointProviderUtils {
         String clientEndpointPath = clientEndpoint.getRawPath();
 
         // [client endpoint path]/[request path]
-        String requestPath = request.getUri().getRawPath();
+        String requestPath = request.encodedPath();
```

Comment:
@dagnir `request.getUri().getRawPath()` should be the same as `encodedPath()` right? This is a micro performance optimization (because there's no need to parse the URI)

Reply:
Yep good catch, it should be: https://github.com/aws/aws-sdk-java-v2/blob/4dd34b1c7ad5a26d0d072c30d3dccda4e1a3aa7b/http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpRequest.java#L157